Mathlib/Algebra/Homology/HomologicalComplex.lean

```diff
@@ -1013,3 +1015,16 @@ theorem mkHom_f_succ_succ (n : ℕ) :
 end MkHom
 
 end CochainComplex
+
+namespace ChainComplex
+
+/-- The Euler characteristic of a chain complex up to index `n`, defined as the alternating sum
+`∑_{k=0}^n (-1)^k rank(C_k)` of the ranks of the chain groups from index 0 to n.
+This is most commonly used when the base ring is a field (where rank means dimension)
+or a principal ideal domain. For a chain complex that is zero above some index,
+choosing `n` to be at least that index gives the total Euler characteristic. -/
+noncomputable def eulerChar {R : Type*} [Ring R] (C : ChainComplex (ModuleCat R) ℕ)
+    (n : ℕ) : ℤ :=
+  ∑ k ∈ Finset.range (n + 1), (-1 : ℤ)^k * Module.finrank R (C.X k)
```

Comment:
This should be moved to a new file.

Reply:
I was a bit uncertain where to put it. I'll put it in its own file.